### ISSUE TITLE
_changes feed not responding for certain since values / overall unhealthy state

### ISSUE THREAD

**Issue by tleyden** [couchbase/sync_gateway#374]
I'm running sync gateway master branch as of 5 days ago, and it's connected to Couchbase Server 2.2 and running under GCE/docker.

When I call the _changes feed for the Todolite database with:

```
curl http://server:port/todolite/_changes?since=3780
```

it works, and returns https://gist.github.com/tleyden/50c25bb06d6255311e90 (server logs included in this gist)

However if I call it with an earlier since, eg:

```
curl http://server:port/todolite/_changes?since=3580
```

it hangs forever.  In the server logs I see:

```
23:42:48.860058 HTTP:  #15879: GET /todolite/_changes?since=3580  (ADMIN)
23:42:48.860343 Changes: MultiChangesFeed({*}, {Since:3580 Limit:0 Conflicts:false IncludeDocs:false Wait:false Continuous:false Terminator:0xc213a97120}) ...
23:42:48.860807 Changes+: MultiChangesFeed: channels expand to channels.TimedSet{"*":0x0} ...
```

I have a feeling that it might be related to Couchbase Server health, because when I go to the Couchbase Server web admin, I see that it's indexing default/_design/sync_gateway:

![screenshot](http://cl.ly/image/1P1G1V1c1F2k/Screen%20Shot%202014-07-19%20at%204.45.28%20PM.png)

and it seems to be stuck there.

This web admin screen shows the node stuck at 100% cpu:

![screenshot](http://cl.ly/image/1R1f1k0Y0b3J/Screen%20Shot%202014-07-19%20at%204.56.20%20PM.png)

Here is the output from running `couchbase-cli server-info`: https://gist.github.com/tleyden/7dc33994433479750349

**Comment by facundofarias** [comment]
Hey.
I am not following this project, but I did find this issue, because we were suffering the same on our production server. After several nights with the service throwing all kind of exceptions (mostly, because of Couchbase timeouts), we've found that the issue was been caused because we were mixing objects JSON and BSON on the same bucket. Somehow the views are trying to index the BSON object, and it stocks there, after some minutes the server does not respond any request anymore. Even restarting the server. The solution was putting the BSON objects on a different bucket. I wanted to comment this here, just in case you think I might be helpful.
